File: Javascript/src/Thermostat.js
function Thermostat() {
	this.temperature = 20;
	this.powerSavingMode = true
}

Thermostat.prototype.increaseTemperature = function() {
	if(this.temperature < this.maximumTemperature()) {
		this.temperature += 1;
	}
};

Thermostat.prototype.decreaseTemperature = function() {
	if(this.temperature > 10) this.temperature -= 1;
};

Thermostat.prototype.maximumTemperature = function() {
	return (this.powerSavingMode) ? 25 : 32;  
};

Thermostat.prototype.resetTemperature = function() {
	this.temperature = 20;
};

Thermostat.prototype.energyUsage = function() {
	if(this.temperature < 18) { 
		return 'efficient'
	} else if (this.temperature < 25) { 
		return 'average'
	} else {
		return 'inefficient'
	}
};

File: Ruby/lib/thermostat.rb
class Thermostat

	def initialize
		@temperature = 20
		@psm_mode = false
	end

	attr_accessor :temperature, :psm_mode

	def turn_psm_on
		@psm_mode = true
	end

	def maximum_temperature
		psm_mode == true ? 25 : 32
	end

	def increase_temperature
		unless temperature >= maximum_temperature
			@temperature += 1
		end
	end

	def decrease_temperature
		unless temperature <= 10
			@temperature -= 1
		end
	end

	def reset_temperature
		@temperature = 20
	end

	def energy_score
		if temperature < 18
			return 'efficient'
		elsif temperature < 25
			return 'average'
		else
			return 'inefficient'
		end
	end

end
